sys/net/network_layer/ng_ipv6/netif/ng_ipv6_netif.c

```diff
@@ -422,15 +439,77 @@ kernel_pid_t ng_ipv6_netif_find_by_prefix(ipv6_addr_t **out, const ipv6_addr_t *
     return res;
 }
 
-static ipv6_addr_t *_match_prefix(kernel_pid_t pid, const ipv6_addr_t *addr,
-                                  bool src_search)
+/**
+ * @brief selects potential source address candidates
+ * see http://tools.ietf.org/html/rfc6724#section-4
+ * @param[in]  iface            the interface used for sending
+ * @param[in]  dest             the destination address
+ * @param[out] candidate_set    a bitfield representing all addresses
+ *                              configured to @p iface, potential candidates
+ *                              will be marked as 1
+ *
+ * @return false if no candidates were found
+ * @return true otherwise
+ *
+ * @pre the interface entry and its set of addresses must not be changed during
+ *      runtime of this function
+ */
+static int _create_candidate_set(ng_ipv6_netif_t *iface,
+                                  const ipv6_addr_t *dest,
+                                  uint8_t *candidate_set)
+{
+    int res = -1;
+
+    DEBUG("gathering candidates\n");
+
+    /* currently this implementation supports only addresses as source address
+     * candidates assigned to this interface. Thus we assume all addresses to be
+     * on interface @p iface */
+    (void) dest;
+
+    for (int i = 0; i < NG_IPV6_NETIF_ADDR_NUMOF; i++) {
+        ng_ipv6_netif_addr_t *iter = &(iface->addrs[i]);
+
+        DEBUG("Checking address: %s\n",
+              ipv6_addr_to_str(addr_str, &(iter->addr), sizeof(addr_str)));
+
+        /* "In any case, multicast addresses and the unspecified address MUST NOT
+         *  be included in a candidate set."
+         */
+        if (ipv6_addr_is_multicast(&(iter->addr)) ||
+            ipv6_addr_is_unspecified(&(iter->addr))) {
+            continue;
+        }
+
+        /* "For all multicast and link-local destination addresses, the set of
+         *  candidate source addresses MUST only include addresses assigned to
+         *  interfaces belonging to the same link as the outgoing interface."
+         *
+         * "For site-local unicast destination addresses, the set of candidate
+         *  source addresses MUST only include addresses assigned to interfaces
+         *  belonging to the same site as the outgoing interface."
+         *  -> we should also be fine, since we're only iterating addresses of
+         *     the sending interface
```

Comment:
The channel has nothing to do here. The sending process has to determine an interface which is the "outgoing interface".